sw/device/tests/aes_interrupt_encryption_test.c

```diff
@@ -0,0 +1,260 @@
+    0x30, 0xc8, 0x1c, 0x46, 0xa3, 0x5c, 0xe4, 0x11, 0xe5, 0xfb, 0xc1,
+    0x19, 0x1a, 0x0a, 0x52, 0xef, 0xf6, 0x9f, 0x24, 0x45, 0xdf, 0x4f,
+    0x9b, 0x17, 0xad, 0x2b, 0x41, 0x7b, 0xe6, 0x6c, 0x37, 0x10};
+
+static const uint8_t kKeyShare1[] = {
+    0x0f, 0x1f, 0x2f, 0x3f, 0x4f, 0x5f, 0x6f, 0x7f, 0x8f, 0x9f, 0xaf,
+    0xbf, 0xcf, 0xdf, 0xef, 0xff, 0x0a, 0x1a, 0x2a, 0x3a, 0x4a, 0x5a,
+    0x6a, 0x7a, 0x8a, 0x9a, 0xaa, 0xba, 0xca, 0xda, 0xea, 0xfa,
+};
+
+OTTF_DEFINE_TEST_CONFIG();
+
+status_t execute_test(void) {
+  // Initialise AES.
+  dif_aes_t aes;
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes));
+  CHECK_DIF_OK(dif_aes_reset(&aes));
+
+  // Mask the key. Note that this should not be done manually. Software is
+  // expected to get the key in two shares right from the beginning.
+
+  uint8_t key_share0[sizeof(kAesModesKey128)];
+  for (int i = 0; i < sizeof(kAesModesKey128); ++i) {
+    key_share0[i] = kAesModesKey128[i] ^ kKeyShare1[i];
+  }
+
+  // "Convert" key share byte arrays to `dif_aes_key_share_t`.
+  dif_aes_key_share_t key;
+  memcpy(key.share0, key_share0, sizeof(key.share0));
+  memcpy(key.share1, kKeyShare1, sizeof(key.share1));
+
+  dif_aes_iv_t iv;
+  memcpy(iv.iv, kAesModesIvCbc, sizeof(iv.iv));
+
+  // Write the initial key share, IV and data in CSRs (known combinations).
+
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeCbc,
+      .key_len = kDifAesKey128,
+      .key_provider = kDifAesKeySoftwareProvided,
+      .mask_reseeding = kDifAesReseedPerBlock,
+      .manual_operation = kDifAesManualOperationAuto,
+      .reseed_on_key_change = false,
+      .ctrl_aux_lock = false,
+  };
+
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, &key, &iv));
+
+  // "Convert" plain data byte arrays to `dif_aes_data_t` array.
+  enum {
+    kAesNumBlocks = 2,
+  };
+
+  dif_aes_data_t plain_text[kAesNumBlocks];
+  dif_aes_data_t cipher_text[kAesNumBlocks];
+
+  memcpy(plain_text[0].data, kAesModesPlainText_block12,
+         sizeof(kAesModesPlainText_block12));
+
+  // Encrypt kAesNumBlocks blocks.
+
+  CHECK_DIF_OK(dif_aes_process_data(&aes, plain_text, cipher_text,
+                                    (size_t)kAesNumBlocks));
+
+  CHECK_DIF_OK(dif_aes_load_data(&aes, plain_text[0]));
+
+  LOG_INFO("Started  AES Encrypt with load function");
+
+  dif_aes_iv_t iv_encrypt1;
+  CHECK_DIF_OK(dif_aes_read_iv(&aes, &iv_encrypt1));
+
+  bool ready = false;
+
+  do {
+    CHECK_DIF_OK(dif_aes_get_status(&aes, kDifAesStatusOutputValid, &ready));
+  } while (!ready);
+
+  // Disable the AES unit to no longer automatically start encryption/decryption
+
+  transaction.manual_operation = kDifAesManualOperationManual;
+
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerDataOutClear));
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerKeyIvDataInClear));
+
+  LOG_INFO(" De-initialized Automatic operation ");
```

Comment:
Nit: why the extra whitespace in the string?

Reply:
Fixed it - and amended the changes!